xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp

```diff
@@ -2427,85 +2357,31 @@ void CAMLCodec::SetVideoRect(const CRect &SrcRect, const CRect &DestRect)
   {
     float xscale = display.Width() / gui.Width();
     float yscale = display.Height() / gui.Height();
-    if (m_stereo_mode == RENDER_STEREO_MODE_SPLIT_VERTICAL)
+    if (m_guiStereoMode == RENDER_STEREO_MODE_SPLIT_VERTICAL)
       xscale /= 2.0;
-    else if (m_stereo_mode == RENDER_STEREO_MODE_SPLIT_HORIZONTAL)
+    else if (m_guiStereoMode == RENDER_STEREO_MODE_SPLIT_HORIZONTAL)
       yscale /= 2.0;
     dst_rect.x1 *= xscale;
     dst_rect.x2 *= xscale;
     dst_rect.y1 *= yscale;
     dst_rect.y2 *= yscale;
   }
 
-  if (m_stereo_mode == RENDER_STEREO_MODE_MONO)
+  if (m_guiStereoMode == RENDER_STEREO_MODE_MONO)
   {
-    std::string mode = GetStereoMode();
-    if (mode == "left_right")
-    {
-      if (!SetVideo3dMode(MODE_3D_TO_2D_L))
-      {
-        // fall back to software scaling if no hw support
-        // was found
-        dst_rect.x2 *= 2.0;
-      }
-    }
-    else if (mode == "right_left")
-    {
-      if (!SetVideo3dMode(MODE_3D_TO_2D_R))
-      {
-        // fall back to software scaling if no hw support
-        // was found
-        dst_rect.x2 *= 2.0;
-      }
-    }
-    else if (mode == "top_bottom")
-    {
-      if (!SetVideo3dMode(MODE_3D_TO_2D_T))
-      {
-        // fall back to software scaling if no hw support
-        // was found
-        dst_rect.y2 *= 2.0;
-      }
-    }
-    else if (mode == "bottom_top")
-    {
-      if (!SetVideo3dMode(MODE_3D_TO_2D_B))
-      {
-        // fall back to software scaling if no hw support
-        // was found
-        dst_rect.y2 *= 2.0;
-      }
-    }
-    else
-      SetVideo3dMode(MODE_3D_DISABLE);
+    std::string videoStereoMode = GetVideoStereoMode();
```

Comment:
Instead calling your function here, call m_processInfo.GetVideoStereoMode here and remove your function.